File: Programowanie_Obiektowe/PO_JAVA/src/Wilk.java
public class Wilk extends Zwierze
{
    public Wilk(int x, int y, Swiat swiat)
    {
        super(Stale.WILK_SILA, Stale.WILK_INICJATYWA, x, y, swiat);
    }

    @Override
    public char rysowanie()
    {
        return Stale.WILK;
    }

    @Override
    public String nazwa()
    {
        return Stale.WILK_S;
    }

    @Override
    public Organizm stworzPotomka(int x, int y)
    {
        return new Wilk(x, y, swiat);
    }
}

File: Programowanie_Obiektowe/PO_C/PO_C/Wilk.cpp
#include "Wilk.h"

using namespace std;

Wilk::Wilk(int x, int y, Swiat& swiat) : Zwierze(WILK_SILA, WILK_INICJATYWA, x, y, swiat) {}

char Wilk::rysowanie() const
{
    return WILK;
}

Organizm* Wilk::stworzPotomka(int x, int y) const
{
    return new Wilk(x, y, swiat);
}